### PR TITLE
Skip test_multi_worker_dataloader_release_pool

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
...to unblock many PR on the CI.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Skip `test_multi_worker_dataloader_release_pool`

### CODE REVIEW

**Comment by leezu** [comment]
Does the test fail on any other platform besides OS X? Currently OS X status test is optional, so failing OS X tests doesn't block PRs

**Comment by haojin2** [comment]
@leezu I still think it's still necessary to skip this for now in case it masks the failures of other tests that comes after this test in the test sequence.

**Comment by leezu** [comment]
Thanks for clarification. Let's disable it. I'm still at loss why this only happens on OS X.
Would be great if anyone with a Mac could help debug this issue.

**Comment by zhreshold** [comment]
I think you can check for mac explicitly in test and disable it rather than skip for all platforms

**Comment by haojin2** [comment]
@zhreshold Good idea, I'll make changes in that way.

**Comment by marcoabreu** [comment]
Can we add a tracking issue in this PR as well as in the code?

File: tests/python/unittest/test_gluon_data.py [hunk 1]
```diff
@@ -269,6 +269,12 @@ def test_multi_worker_dataloader_release_pool():
     if os.name == 'nt':
         print('Skip for windows since spawn on windows is too expensive.')
         return
+
+    from sys import platform
+    if platform == 'darwin':
+        print('Skip for MacOS due to flaky failures')
```
Comment (leezu) [inline tests/python/unittest/test_gluon_data.py]:
```suggestion
        print('Skip for MacOS due to https://github.com/apache/incubator-mxnet/issues/17782')
```

Comment (leezu) [reply tests/python/unittest/test_gluon_data.py]: @haojin2 can you merge the suggested change or something similar to address Marco's concern?

**Pull request merged**